### PR TITLE
Add SymmetryTransformerGenerator, with plane symmetry only currently

### PR DESCRIPTION
**Pull request by RocksSavage**
Beware, the option to optionally use StitchedMeshGenerator has not been tested. Tests have not been written.
I would appreciate it if the test currently in simple.i would end up in the documentation as an example of what
default parameters are ran when the stitch option is exercised.

- [x] add tests
- [x] add stitching capability
- [x] document extent of support in md file

Refs #21578

### CODE REVIEW

**Comment by GiudGiud** [comment]
This is on hold until we can get better fixing of negative volumes in libmesh

**Comment by GiudGiud** [comment]
This will be near-ready on the next libmesh submodule update.

**Comment by GiudGiud** [comment]
@loganharbour this should be ready for review please

**Comment by GiudGiud** [comment]
nevermind still no libmesh update

**Comment by GiudGiud** [comment]
@loganharbour ready for review.

I've had to add a bit of lower quality code (const_cast) to allow for a sub-generator to run after a main generator. 
If that's not agreeable, then I reckon I want to delete the stitching and get this merged.

File: framework/src/meshgenerators/MeshGenerator.C [hunk 1]
```diff
@@ -138,12 +138,22 @@ MeshGenerator::generateInternal()
...
-  return this->getMeshByName(name);
+  // getMeshByName will add the dependency
+  if (sub_executes_before)
+    return this->getMeshByName(name);
+  else
+  {
+    const_cast<MeshGenerator &>(_app.getMeshGenerator(name))
+        .getDependencies()
+        .push_back(MooseObject::name());
+    return _app.getMeshGeneratorOutput(name);
```
Comment (loganharbour) [inline framework/src/meshgenerators/MeshGenerator.C:157]: I've got a cleaner way to do this - I'll work on it this morning

File: framework/src/meshgenerators/SymmetryTransformGenerator.C [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  // enforce 3D coordinates
+  if (_mirror_point_vector.size() != 3)
+    mooseError("mirror_point should be a 3d vector, but only ",
+               _mirror_point_vector.size(),
+               "components were specified.");
+
+  if (_mirror_normal_vector.size() != 3)
+    mooseError(" mirror_normal_vector should be a 3d vector, but only ",
+               _mirror_normal_vector.size(),
+               "components were specified.");
```
Comment (loganharbour) [inline framework/src/meshgenerators/SymmetryTransformGenerator.C:57]: paramErrors

File: framework/include/meshgenerators/SymmetryTransformGenerator.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  std::unique_ptr<MeshBase> & _input;
+
+  /// A point in the mirror plane from which to transform the mesh
+  RealEigenVector _mirror_point_vector;
+
+  /// The normal of the mirror plane from which to transform the mesh
+  RealEigenVector _mirror_normal_vector;
+
+  /// The boundaries to stitch together
+  std::vector<std::vector<std::string>> _stitch_boundaries_pairs;
```
Comment (loganharbour) [inline framework/include/meshgenerators/SymmetryTransformGenerator.h:37]: Const where possible

**Review by loganharbour — CHANGES_REQUESTED** [review]
I'm really against the subgenerator for a few reasons:

1. The implementation of the before action is poor. This is resolvable and thus is minor.
2. I think that it will be confusing if we allow the addition of subgens like this. We could end up in a position where we could end up with a wonky graph that isn't intuitive to debug. 
3. The action of the stitching is so simple and only requires copying one input to another in another MG. I think it's a poor use of the subgenerator system. It makes it look like we're okay with having a ton of meta generators - which I think is a poor idea because the best part of the MG system is its pluggability, and having to maintain all of that meta interaction sounds awful.

File: framework/include/meshgenerators/SymmetryTransformGenerator.h (hunk 3 above)
Comment (GiudGiud) [reply framework/include/meshgenerators/SymmetryTransformGenerator.h]: should be max now

**Comment by GiudGiud** [comment]
Should be good now

**Review by loganharbour — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the contribution @RocksSavage !
I hope all is well in Utah

**Pull request merged**